### PR TITLE
Editing UI for Raw HTML embedding feature

### LINKED ISSUE #8291: Improve UI for HTML embed feature
https://github.com/ckeditor/ckeditor5/pull/8290 provides a poor UI and the working editing part. Now we need to improve the UI of the feature.

Mock-ups / design proposals: https://github.com/ckeditor/ckeditor5/issues/8204#issuecomment-712120055.

### PR DESCRIPTION
**Pull request by pomek**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Internal (html-embed): Editing UI for Raw HTML embedding feature. Closes #8291.

Tests (theme-lark): Added new icons to the manual test.

---

### Additional information

~Requires: https://github.com/ckeditor/ckeditor5/pull/8290.~

### CODE REVIEW

**Comment by pomek** [comment]
Embedding a manual test HTML into the editor:

image

**Comment by Reinmar** [comment]
Let's remember to review the lang strings: https://github.com/ckeditor/ckeditor5/pull/8290/files#diff-173c07d9b0cbd6a3480b12c92fcef0e7882d810c50455d86c0192539dcc4561a.

**Comment by oleq** [comment]
The widget needs some overflow:

File: packages/ckeditor5-html-embed/tests/htmlembedediting.js [hunk 1]
```diff
@@ -321,36 +322,115 @@ describe( 'HtmlEmbedEditing', () => {
...
+
 				// Switch to preview mode.
-				toggleIconElement.getCustomProperty( 'domElement' ).click();
+				toggleButton.click();
 
-				const sourceElement = viewHtmlContainer.getChild( 1 );
+				expect( viewHtmlContainer.getChild( 1 ).getCustomProperty( 'domElement' ).disabled ).to.equal( true );
+			} );
+
+			it( 'should disable the source element after clicking the toggle button when edit source mode is enabled', () => {
```
Comment (pomek) [inline packages/ckeditor5-html-embed/tests/htmlembedediting.js:373]:
```suggestion
			it( 'should restore the toggle button icon after leaving the "edit source mode"', () => {
```

**Comment by oleq** [comment]
The button needs some sane `z-index` (it's underneath the preview)

**Comment by Reinmar** [comment]
Just checking – do we have a mask over the preview to intercept all interactions? IMO, clicking the entire area should result in selecting the widget with one exception - clicking the "edit" button should work.

**Comment by pomek** [comment]
> Just checking – do we have a mask over the preview to intercept all interactions?

No, we don't.

**Comment by pomek** [comment]
Switching the selected element after inserting an empty `rawHtml` element.

image

The container needs a `min-height` value.

**Comment by pomek** [comment]
Two widgets nearby themselves look bad:

image

**Comment by pomek** [comment]
@Reinmar, https://github.com/ckeditor/ckeditor5/pull/8301#issuecomment-713395818, fixed in 14e13fa.

**Comment by oleq** [comment]
Some suggestions concerning CSS classes according to the [code style guide](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/code-style.html#css-classes):

*   raw-html -> ✅
*   raw-html--preview-enabled -> raw-html\_preview\_enabled
*   raw-html--display-preview -> raw-html\_preview\_visible
*   raw-html\_\_switch-mode -> raw-html\_\_mode-toggle-button
*   raw-html\_\_source -> ✅
*   raw-html\_\_preview -> ✅

**Comment by pomek** [comment]
@oleq, names of the CSS classes are fixed in e89b76e.

**Comment by pomek** [comment]
After clicking a textarea, the model selection should be set on the `rawHtml` element. If not, previews will not work or the command will update the wrong element.

**Comment by pomek** [comment]
Commit build:

image

PR build:

image

**Pull request merged**